File: Level-2/81.java
import java.util.*;

class Solution {
    public int solution(int[] order) {
        int answer = 0;
        
        Stack<Integer> stack = new Stack<>();
        
        int idx = 0;
        for(int i = 1; i <= order.length; i++) {
            if(order[idx] != i) {
                stack.add(i);
                continue;
            }
            answer++;
            idx++;
            
            while(stack.size() != 0 && order[idx] == stack.peek()) {
                answer++;
                idx++;
                stack.pop();
            }
        }
        
        return answer;
    }
}

File: Level-0/81.js
function solution(n) {
    let answer = [];
    for(let i = 1; i <= Math.sqrt(n); i++) {
        if(n % i === 0) {
            if(i === n / i) {
                answer.push(i);
            }else {
                answer.push(i);
                answer.push(n / i);
            }
        }
    }
    answer.sort((a, b) => a - b);
    return answer;
}
